### DISCUSSION TITLE
Ocular does not save any data

### DISCUSSION THREAD

**Discussion by rhasselbusch** [simonwep/ocular#50] [General]
Hi together,

i try to host Ocular via docker.
I'm new to Docker, so it's likely that i missed somethin in my Compose file.

This is my Docker Compose file. Sorry for posting it without a Code block but the Code block seems to break the formatting.

services:
  backend:
    image: ghcr.io/simonwep/genesis:v1.2
    restart: unless-stopped
    volumes:
      - ./data:/app/.data
    environment:
      - GENESIS_PORT
      - GENESIS_DB_PATH
      - GENESIS_CREATE_USERS
      - GENESIS_AUTHORIZED_URIS
      - GENESIS_JWT_SECRET
      - GENESIS_JWT_TOKEN_EXPIRATION
      - GENESIS_JWT_COOKIE_ALLOW_HTTP
      - GENESIS_USERNAME_PATTERN
      - GENESIS_KEY_PATTERN
      - GENESIS_DATA_MAX_SIZE
      - GENESIS_KEYS_PER_USER
      - GENESIS_GIN_MODE
      - GENESIS_LOG_MODE
  frontend:
    image: ghcr.io/simonwep/ocular:v1.5
    restart: unless-stopped
  nginx:
    image: nginx:1.24-alpine
    restart: unless-stopped
    ports:
      - 7363:80
    volumes:
      - /docker-centralstore/ocular/config/nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - backend
      - frontend

And this is the .env file (Comments are escaped)

`# Database location
GENESIS_DB_PATH=.data

`# JWT secret known only to your token generator
GENESIS_JWT_SECRET=6f8c47a6b1e2d3f9a1c2e3b4d5f6a7b8c9d0e1f2a3b4c5d6e7f8g9h0a1b2c3d3

`# JWT expiration in minutes
GENESIS_JWT_TOKEN_EXPIRATION=120960

`# If the session cookie for the backend should be allowed to be sent over http
`# Dangerous, it's best to run it behind a reverse proxy with https
GENESIS_JWT_COOKIE_ALLOW_HTTP=false

`# Gin mode, either test, release or debug
GENESIS_GIN_MODE=release

`# Zap loggger, either production or development
GENESIS_LOG_MODE=production

`# Port to listen on, leave it at 80 if you're using a reverse proxy
GENESIS_PORT=80

`# Base url to listen for requests
GENESIS_BASE_URL=https://money.someurl.com

`# Use ! as suffix for the username to indicate that this user
`# should be created as an admin. These can add, remove and edit users.
GENESIS_CREATE_USERS=admin!:SuperSecretPassword

`# Allowed username pattern
GENESIS_USERNAME_PATTERN=^[\w]{0,32}$

`# Allowed key pattern
GENESIS_KEY_PATTERN=^[\w]{0,32}$

`# Maximum size of each key in kilobytes
GENESIS_DATA_MAX_SIZE=512

`# Maximum amount of datasets per user
GENESIS_KEYS_PER_USER=2

The containers are starting fine and i can access the frontend and enter some data, change the language and currency. When i reload the page all data / changes are gone, so i assume that i misonfigured anything and the frontend is not communicating with the backend.

Can anyone with a deeper understanding of Ocular/Docker point me in the right direction?

best regards
Rolf

**Comment by simonwep** [comment]
Sorry for coming back to this so late, I totally missed out on discussion. Are you still having problems? If so could you open an issue over at https://github.com/simonwep/ocular-docker to further discuss it? Others also had problems, maybe yours is already solved there!

Closing it for now.
